### ISSUE TITLE
NameError from protorpc happen in 4.12

### ISSUE THREAD

**Issue by imdreamrunner** [GoogleCloudPlatform/gsutil#273]
When I run the `gsutil` command installed using `pip` on Mac, I got the following exception.

```
Traceback (most recent call last):
  File "/usr/local/bin/gsutil", line 9, in <module>
    load_entry_point('gsutil==4.12', 'console_scripts', 'gsutil')()
  File "/Library/Python/2.7/site-packages/pkg_resources/__init__.py", line 552, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Library/Python/2.7/site-packages/pkg_resources/__init__.py", line 2672, in load_entry_point
    return ep.load()
  File "/Library/Python/2.7/site-packages/pkg_resources/__init__.py", line 2345, in load
    return self.resolve()
  File "/Library/Python/2.7/site-packages/pkg_resources/__init__.py", line 2351, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Library/Python/2.7/site-packages/gslib/__main__.py", line 53, in <module>
    from gslib import wildcard_iterator
  File "/Library/Python/2.7/site-packages/gslib/wildcard_iterator.py", line 37, in <module>
    from gslib.translation_helper import GenerationFromUrlAndString
  File "/Library/Python/2.7/site-packages/gslib/translation_helper.py", line 25, in <module>
    from apitools.base.py import encoding
  File "/Library/Python/2.7/site-packages/apitools/base/py/__init__.py", line 5, in <module>
    from apitools.base.py.base_api import *
  File "/Library/Python/2.7/site-packages/apitools/base/py/base_api.py", line 10, in <module>
    from protorpc import message_types
  File "/Library/Python/2.7/site-packages/protorpc/message_types.py", line 28, in <module>
    from . import messages
  File "/Library/Python/2.7/site-packages/protorpc/messages.py", line 398, in <module>
    class Enum(six.with_metaclass(_EnumClass, object)):
  File "/Library/Python/2.7/site-packages/protorpc/messages.py", line 300, in __init__
    if not (bases == (object,) or bases == (Enum,)):
NameError: global name 'Enum' is not defined
```

Versions:
gsutil 4.12
protorpc 0.10.0

**Comment by mfschwartz** [comment]
Hi - can you please try updating the version of six that you have installed? I believe that updating to the latest (1.9.0) version should fix this problem. Thanks.

Mike

**Comment by mfschwartz** [comment]
Before you update can you please let us know what version of six you have installed? One of my teammates pointed out that six 1.8.0 should have the required fix. You can do this as follows:

% python
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.__version__
'1.9.0'
>>> 

Thanks,

Mike

**Comment by imdreamrunner** [comment]
Thanks for your reply and research!

What's happening is interesting in my computer.

I try to check the installed package `six`'s status on pip, and I get

```
% pip show six
---
Metadata-Version: 2.0
Name: six
Version: 1.9.0
Summary: Python 2 and 3 compatibility utilities
Home-page: http://pypi.python.org/pypi/six/
Author: Benjamin Peterson
Author-email: benjamin@python.org
License: MIT
Location: /Library/Python/2.7/site-packages
Requires: 
```

However, if I import it in python, I could not get its `version` attribute.

```
% python
Python 2.7.6 (default, Sep  9 2014, 15:04:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.version
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'version'
```

Thanks to pointing out the problem of `six` and I will also look into that.

**Comment by mfschwartz** [comment]
Hi - what happens if instead of "six.version" you enter "six.__version__" in the python interpreter?

**Comment by mfschwartz** [comment]
Hi - I just realized that markdown interpreted what I entered as formatting. Let me try again:

Please try entering

six.\__version\__

(that's six dot two-underscores version two-underscores)

**Comment by craigcitro** [comment]
one more note from the sidelines: if `import pip; pip.__version__` ends up showing a different version than `pip show six`, it's likely an issue with either multiple `pip`s or `pip` installing somewhere at the back of `sys.path`.

**Comment by craigcitro** [comment]
@imdreamrunner so i think we've figured out what's going on: you've got an older `six` managed by `easy_install`, which is shadowing the newer copy of `six` installed by `pip`.

another user hit this, and doing `sudo easy_install -U six` fixed the issue.

**Comment by imdreamrunner** [comment]
Thank you @craigcitro 
You are correct!

`sudo easy_install -U six` fixed this issue.

**Comment by neeln4j** [comment]
Thank you guys for bring up this issue here , i had this issue now its resolved.  can anyone explain "sudo easy_install -U six" what this command will do

**Comment by mfschwartz** [comment]
"sudo easy_install -U six" will install the package "six" on your system, which is a Python 2 and 3 compatibility library (see https://pypi.python.org/pypi/six)

**Comment by neeln4j** [comment]
@mfschwartz  thanks mfschwartz

**Comment by dheeraj143** [comment]
@craigcitro Thank you so much! `sudo easy_install -U six` helped me!

**Comment by ayush--s** [comment]
issue still persists for me. 

```bash
ayush:/code/ $ gsutil ls gs://uspto-pair/applications/0800401*
Traceback (most recent call last):
  File "/usr/local/bin/gsutil", line 9, in <module>
    load_entry_point('gsutil==4.16', 'console_scripts', 'gsutil')()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 558, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2682, in load_entry_point
    return ep.load()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2355, in load
    return self.resolve()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2361, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.4/dist-packages/gslib/__init__.py", line 31, in <module>
    import gslib.exception
  File "/usr/local/lib/python3.4/dist-packages/gslib/exception.py", line 30, in <module>
    class AbortException(StandardError):
NameError: name 'StandardError' is not defined
```

```python
>>> import six

>>> six.__version__
'1.10.0'
```
